containers/src/Kokkos_DualView.hpp

```diff
@@ -1186,12 +1201,11 @@ void deep_copy(
   }
 }
 
-template <class ExecutionSpace, class DT, class DL, class DD, class DM,
-          class ST, class SL, class SD, class SM>
+template <class ExecutionSpace, class DT, class... DP, class ST, class... SP>
 void deep_copy(
     const ExecutionSpace& exec,
-    DualView<DT, DL, DD, DM> dst,  // trust me, this must not be a reference
-    const DualView<ST, SL, SD, SM>& src) {
+    DualView<DT, DP...>& dst,
```

Comment:
I would prefer
```suggestion
    const DualView<DT, DP...>& dst,
```
to mirror the overloads for `Kokkos::View` more closely.

Reply:
I'm reverting this because these functions are calling `modify_host` on `dst` that mustn't be `const`:
```bash
In file included from /__w/kokkos/kokkos/containers/unit_tests/TestDualView.hpp:25:
/__w/kokkos/kokkos/containers/unit_tests/../src/Kokkos_DualView.hpp:1194:9: error: no matching member function for call to 'modify_host'
    dst.modify_host();
    ~~~~^~~~~~~~~~~
/__w/kokkos/kokkos/containers/unit_tests/TestDualView.hpp:185:13: note: in instantiation of function template specialization 'Kokkos::deep_copy<int **, Kokkos::LayoutLeft, Kokkos::OpenMP, int **, Kokkos::LayoutLeft, Kokkos::OpenMP>' requested here
    Kokkos::deep_copy(b, a);
```